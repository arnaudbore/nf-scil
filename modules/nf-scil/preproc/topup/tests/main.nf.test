// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test preproc/topup
nextflow_process {

    name "Test Process PREPROC_TOPUP"
    script "../main.nf"
    process "PREPROC_TOPUP"

    tag "modules"
    tag "modules_nfcore"
    tag "preproc"
    tag "preproc/topup"

    tag "subworflows"
    tag "subworflows/load_test_data"

    config "./nextflow.config"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "topup_eddy.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("topup") {
        when {
            process {


                """
             input[0] = LOAD_DATA.out.test_data_directory
                 .map{ test_data_directory -> [
                 [ id:'test', single_end:false ], // meta map
                 file("\${test_data_directory}/sub-01_dir-AP_dwi.nii.gz", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-AP_dwi.bval", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-AP_dwi.bvec", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-AP_sbref.nii.gz", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-PA_dwi.nii.gz", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-PA_dwi.bval", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-PA_dwi.bvec", checkIfExists: true),
                 file("\${test_data_directory}/sub-01_dir-PA_sbref.nii.gz", checkIfExists: true)
             ]}
             input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
