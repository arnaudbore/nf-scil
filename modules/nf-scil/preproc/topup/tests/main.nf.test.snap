{
    "topup": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__corrected_b0s.nii:md5,df45f8ec14b11f3f9abcebbfc3e874c0"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "topup_results_fieldcoef.nii:md5,acf4f66cd5d2d4a918da04a5f43d22c6"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "topup_results_movpar.txt:md5,57542f26425b0842154410bda572c099"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__rev_b0_warped.nii.gz:md5,64d2123d221d60d8b34d07ef8283c367"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__rev_b0_mean.nii.gz:md5,a0c63d7a8a234d85eeb63045ae2a8f60"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_mean.nii.gz:md5,f50e652bcdd1d45f7a4c592df7eb851a"
                    ]
                ],
                "6": [
                    "versions.yml:md5,9e086ab052cde49b01096c3bc8f524c0"
                ],
                "b0_mean": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__b0_mean.nii.gz:md5,f50e652bcdd1d45f7a4c592df7eb851a"
                    ]
                ],
                "rev_b0_mean": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__rev_b0_mean.nii.gz:md5,a0c63d7a8a234d85eeb63045ae2a8f60"
                    ]
                ],
                "rev_b0_warped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__rev_b0_warped.nii.gz:md5,64d2123d221d60d8b34d07ef8283c367"
                    ]
                ],
                "topup_corrected_b0s": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__corrected_b0s.nii:md5,df45f8ec14b11f3f9abcebbfc3e874c0"
                    ]
                ],
                "topup_fieldcoef": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "topup_results_fieldcoef.nii:md5,acf4f66cd5d2d4a918da04a5f43d22c6"
                    ]
                ],
                "topup_movpart": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "topup_results_movpar.txt:md5,57542f26425b0842154410bda572c099"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9e086ab052cde49b01096c3bc8f524c0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T21:00:19.157707"
    }
}